src/improvements/template/OPCMUpgradeV220toV400.sol

```diff
@@ -0,0 +1,305 @@
+
+        // === OPCM for U14 and U15 ===
+        OPCM_V300 = tomlContent.readAddress(".addresses.OPCMUpgradeV300");
+        OPCM_V300_PRESTATE = tomlContent.readBytes32(".OPCMUpgradeV300_PRESTATE");
+        OPCM_V300_UPDATE_PRESTATE = tomlContent.readBytes32(".OPCMUpgradeV300_UPDATE_PRESTATE");
+        OPCM_TARGETS.push(OPCM_V300);
+        require(IOPContractsManager(OPCM_V300).version().eq("1.9.0"), "Incorrect OPCM - expected version 1.9.0");
+        vm.label(OPCM_V300, "OPCMUpgradeV300");
+
+        // === OPCM for U16 ===
+        OPCM_V400 = tomlContent.readAddress(".addresses.OPCMUpgradeV400");
+        OPCM_TARGETS.push(OPCM_V400);
+        require(IOPContractsManager(OPCM_V400).version().eq("2.4.0"), "Incorrect OPCM - expected version 2.4.0");
+        vm.label(OPCM_V400, "OPCMUpgradeV400");
+
+        // === Standard Validator for U13 ===
+        STANDARD_VALIDATOR_V200 = IStandardValidatorV200(tomlContent.readAddress(".addresses.StandardValidatorV200"));
+        require(address(STANDARD_VALIDATOR_V200).code.length > 0, "ValidatorV200 not deployed");
+        vm.label(address(STANDARD_VALIDATOR_V200), "StandardValidatorV200");
+
+        // === Standard Validator for U14 and U15 ===
+        STANDARD_VALIDATOR_V300 = IStandardValidatorV300(tomlContent.readAddress(".addresses.StandardValidatorV300"));
+        require(address(STANDARD_VALIDATOR_V300).code.length > 0, "ValidatorV300 not deployed");
+        vm.label(address(STANDARD_VALIDATOR_V300), "StandardValidatorV300");
+
+        // === Standard Validator for U16 ===
+        STANDARD_VALIDATOR_V400 = IStandardValidatorV400(tomlContent.readAddress(".addresses.StandardValidatorV400"));
+        require(address(STANDARD_VALIDATOR_V400).code.length > 0, "ValidatorV400 not deployed");
+        vm.label(address(STANDARD_VALIDATOR_V400), "StandardValidatorV400");
+    }
+
+    /// @notice Performs the atomic upgrade (U12 to U16) for all chains
+    function _build(address) internal override {
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        IOPContractsManager.OpChainConfig[] memory opChainConfigs =
+            new IOPContractsManager.OpChainConfig[](chains.length);
+
+        // === Upgrade to U13 ===
+        for (uint256 i = 0; i < chains.length; i++) {
+            opChainConfigs[i] = IOPContractsManager.OpChainConfig({
+                systemConfigProxy: ISystemConfig(superchainAddrRegistry.getAddress("SystemConfigProxy", chains[i].chainId)),
+                proxyAdmin: IProxyAdmin(superchainAddrRegistry.getAddress("ProxyAdmin", chains[i].chainId)),
+                absolutePrestate: Claim.wrap(OPCM_V220_PRESTATE)
+            });
+        }
+
+        (bool success1,) = OPCM_V220.delegatecall(abi.encodeCall(IOPContractsManager.upgrade, (opChainConfigs)));
+        require(success1, "OPCMUpgradeV220: upgrade call failed in _build.");
+
+        // === Validator for U13 ===
```

Comment:
I've commented out the validations in this function. I think we will still see errors and we should assert on those. Of course we will need to make sure that the errors we do see are benign and expected.